lib/fizzy/execute.cpp

```diff
@@ -150,11 +150,29 @@ inline uint64_t popcnt64(uint64_t value) noexcept
 
 Instance instantiate(const Module& module)
 {
-    // FIXME: set pages from proper section
-    constexpr unsigned pages = 1;
+    size_t memory_min, memory_max;
+    if (module.memorysec.size() > 1)
+    {
+        // FIXME: better error handling
+        throw std::runtime_error("Cannot support more than 1 memory section.");
+    }
+    else if (module.memorysec.size() > 0)
+    {
+        memory_min = module.memorysec[0].limits.min;
+        memory_max = module.memorysec[0].limits.max;
+        // FIXME: better error handling
+        if (memory_max > (memorylimit / pagesize))
```

Comment:
seems it was simpler to define `memorylimit` in pages

Reply:
Yes, but since this is part instantiation I think it's fine for simplicity (e.g. it is clear the VM will only use up to 256MB memory)